oauth.go

```diff
@@ -99,39 +117,57 @@ func (h oauthHandler) viewOauthInit(app *App, w http.ResponseWriter, r *http.Req
 
 func configureSlackOauth(parentHandler *Handler, r *mux.Router, app *App) {
 	if app.Config().SlackOauth.ClientID != "" {
+		var stateRegisterClient *oauthStateRegisterer = nil
+		if app.Config().SlackOauth.StateRegisterLocation != "" {
+			stateRegisterClient = &oauthStateRegisterer{
+				location:         app.Config().SlackOauth.StateRegisterLocation,
+				callbackLocation: app.Config().App.Host + "/oauth/callback",
+				httpClient:       config.DefaultHTTPClient(),
+			}
+		}
 		oauthClient := slackOauthClient{
-			ClientID:         app.Config().SlackOauth.ClientID,
-			ClientSecret:     app.Config().SlackOauth.ClientSecret,
-			TeamID:           app.Config().SlackOauth.TeamID,
-			CallbackLocation: app.Config().App.Host + "/oauth/callback",
-			HttpClient:       config.DefaultHTTPClient(),
+			ClientID:     app.Config().SlackOauth.ClientID,
+			ClientSecret: app.Config().SlackOauth.ClientSecret,
+			TeamID:       app.Config().SlackOauth.TeamID,
+			HttpClient:   config.DefaultHTTPClient(),
+			//CallbackLocation: app.Config().App.Host + "/oauth/callback",
+			CallbackLocation: "http://localhost:5000/callback",
```

Comment:
Can we make this endpoint configurable somehow? Maybe in `SlackOauth`, and falling back to the local endpoint if there's no config value?